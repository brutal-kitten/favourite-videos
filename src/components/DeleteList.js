import React, { Component } from 'react';

class DeleteList extends Component {

  //calls function to delete whole list
  handleClick = (event) => {

    event.preventDefault();
    this.props.deleteList();
  }


  render() {

    return(
      <div className="deleteList" >
        <button className="buttonDeleteList" type="button" onClick={(event) => this.handleClick(event)}>Delete list</button>
      </div>
    )
  }
}

export default DeleteList;
