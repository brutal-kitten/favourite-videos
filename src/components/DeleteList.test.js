import React from 'react';
import ReactDOM from 'react-dom';
import DeleteList from './DeleteList.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <DeleteList />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleclick function after click', () => {
  const mockDeleteList = jest.fn();
  const props = {
    deleteList: mockDeleteList
  };

  const wrapper = shallow(<DeleteList {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  it('should call deleteList function', () => {
    expect(mockDeleteList.mock.calls.length).toBe(1);
  });
})
