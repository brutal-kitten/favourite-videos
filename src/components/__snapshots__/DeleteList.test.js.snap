// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="deleteList"
>
  <button
    className="buttonDeleteList"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    Delete list
  </button>
</div>
`;
